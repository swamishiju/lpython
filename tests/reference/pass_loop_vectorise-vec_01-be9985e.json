{
    "basename": "pass_loop_vectorise-vec_01-be9985e",
    "cmd": "lpython --pass=loop_vectorise --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/vec_01.py",
    "infile_hash": "f85ca108780c53c54878d119822d56fb834cf4b5121511cbaca2c2fe",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "pass_loop_vectorise-vec_01-be9985e.stdout",
    "stdout_hash": "21725daa434a5227fed9ca502679acad0a7ef61c6778f89dbe3bf906",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}