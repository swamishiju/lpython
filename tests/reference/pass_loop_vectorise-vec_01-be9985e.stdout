(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        227
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [loop_vec]
                                    []
                                    [(SubroutineCall
                                        2 loop_vec
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            loop_vec:
                                (Function
                                    (SymbolTable
                                        223
                                        {
                                            a:
                                                (Variable
                                                    223
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 9216 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            b:
                                                (Variable
                                                    223
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 9216 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    223
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            vector_copy_f64[9216]f64[9216]i32@IntrinsicOptimization:
                                                (Function
                                                    (SymbolTable
                                                        229
                                                        {
                                                            __1_k:
                                                                (Variable
                                                                    229
                                                                    __1_k
                                                                    []
                                                                    Local
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 4)
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                    .false.
                                                                    .false.
                                                                    ()
                                                                    .false.
                                                                    .false.
                                                                ),
                                                            arg0:
                                                                (Variable
                                                                    229
                                                                    arg0
                                                                    []
                                                                    In
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Array
                                                                        (Real 8)
                                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                                        (IntegerConstant 9216 (Integer 4) Decimal))]
                                                                        FixedSizeArray
                                                                    )
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                    .false.
                                                                    .false.
                                                                    ()
                                                                    .false.
                                                                    .false.
                                                                ),
                                                            arg1:
                                                                (Variable
                                                                    229
                                                                    arg1
                                                                    []
                                                                    In
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Array
                                                                        (Real 8)
                                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                                        (IntegerConstant 9216 (Integer 4) Decimal))]
                                                                        FixedSizeArray
                                                                    )
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                    .false.
                                                                    .false.
                                                                    ()
                                                                    .false.
                                                                    .false.
                                                                ),
                                                            arg2:
                                                                (Variable
                                                                    229
                                                                    arg2
                                                                    []
                                                                    In
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 4)
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                    .false.
                                                                    .false.
                                                                    ()
                                                                    .false.
                                                                    .false.
                                                                ),
                                                            arg3:
                                                                (Variable
                                                                    229
                                                                    arg3
                                                                    []
                                                                    In
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 4)
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                    .false.
                                                                    .false.
                                                                    ()
                                                                    .false.
                                                                    .false.
                                                                ),
                                                            arg4:
                                                                (Variable
                                                                    229
                                                                    arg4
                                                                    []
                                                                    In
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 4)
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                    .false.
                                                                    .false.
                                                                    ()
                                                                    .false.
                                                                    .false.
                                                                ),
                                                            arg5:
                                                                (Variable
                                                                    229
                                                                    arg5
                                                                    []
                                                                    In
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 4)
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                    .false.
                                                                    .false.
                                                                    ()
                                                                    .false.
                                                                    .false.
                                                                )
                                                        })
                                                    vector_copy_f64[9216]f64[9216]i32@IntrinsicOptimization
                                                    (FunctionType
                                                        [(Array
                                                            (Real 8)
                                                            [((IntegerConstant 0 (Integer 4) Decimal)
                                                            (IntegerConstant 9216 (Integer 4) Decimal))]
                                                            FixedSizeArray
                                                        )
                                                        (Array
                                                            (Real 8)
                                                            [((IntegerConstant 0 (Integer 4) Decimal)
                                                            (IntegerConstant 9216 (Integer 4) Decimal))]
                                                            FixedSizeArray
                                                        )
                                                        (Integer 4)
                                                        (Integer 4)
                                                        (Integer 4)
                                                        (Integer 4)]
                                                        ()
                                                        Source
                                                        Implementation
                                                        ()
                                                        .false.
                                                        .false.
                                                        .false.
                                                        .false.
                                                        .false.
                                                        []
                                                        .false.
                                                    )
                                                    []
                                                    [(Var 229 arg0)
                                                    (Var 229 arg1)
                                                    (Var 229 arg2)
                                                    (Var 229 arg3)
                                                    (Var 229 arg4)
                                                    (Var 229 arg5)]
                                                    [(Assignment
                                                        (Var 229 __1_k)
                                                        (IntegerBinOp
                                                            (Var 229 arg2)
                                                            Sub
                                                            (Var 229 arg4)
                                                            (Integer 4)
                                                            ()
                                                        )
                                                        ()
                                                        .false.
                                                    )
                                                    (WhileLoop
                                                        ()
                                                        (IntegerCompare
                                                            (IntegerBinOp
                                                                (Var 229 __1_k)
                                                                Add
                                                                (Var 229 arg4)
                                                                (Integer 4)
                                                                ()
                                                            )
                                                            Lt
                                                            (Var 229 arg3)
                                                            (Logical 4)
                                                            ()
                                                        )
                                                        [(Assignment
                                                            (Var 229 __1_k)
                                                            (IntegerBinOp
                                                                (Var 229 __1_k)
                                                                Add
                                                                (Var 229 arg4)
                                                                (Integer 4)
                                                                ()
                                                            )
                                                            ()
                                                            .false.
                                                        )
                                                        (Assignment
                                                            (ArrayItem
                                                                (Var 229 arg0)
                                                                [(()
                                                                (Var 229 __1_k)
                                                                ())]
                                                                (Real 8)
                                                                RowMajor
                                                                ()
                                                            )
                                                            (ArrayItem
                                                                (Var 229 arg1)
                                                                [(()
                                                                (Var 229 __1_k)
                                                                ())]
                                                                (Real 8)
                                                                RowMajor
                                                                ()
                                                            )
                                                            ()
                                                            .false.
                                                        )]
                                                        []
                                                    )]
                                                    ()
                                                    Public
                                                    .false.
                                                    .false.
                                                    ()
                                                )
                                        })
                                    loop_vec
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Assignment
                                        (Var 223 a)
                                        (ArrayBroadcast
                                            (ArrayItem
                                                (Var 223 a)
                                                [(()
                                                (IntegerConstant 0 (Integer 4) Decimal)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayConstant
                                                4
                                                [9216]
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 1 (Integer 4) Decimal)
                                                    (IntegerConstant 1 (Integer 4) Decimal))]
                                                    FixedSizeArray
                                                )
                                                ColMajor
                                            )
                                            (Array
                                                (Real 8)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 9216 (Integer 4) Decimal))]
                                                FixedSizeArray
                                            )
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (Var 223 b)
                                        (ArrayBroadcast
                                            (ArrayItem
                                                (Var 223 b)
                                                [(()
                                                (IntegerConstant 0 (Integer 4) Decimal)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayConstant
                                                4
                                                [9216]
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 1 (Integer 4) Decimal)
                                                    (IntegerConstant 1 (Integer 4) Decimal))]
                                                    FixedSizeArray
                                                )
                                                ColMajor
                                            )
                                            (Array
                                                (Real 8)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 9216 (Integer 4) Decimal))]
                                                FixedSizeArray
                                            )
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 223 i)
                                        (IntegerConstant 0 (Integer 4) Decimal)
                                        (IntegerBinOp
                                            (IntegerConstant 9216 (Integer 4) Decimal)
                                            Sub
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            (Integer 4)
                                            (IntegerConstant 9215 (Integer 4) Decimal)
                                        )
                                        (IntegerConstant 1 (Integer 4) Decimal))
                                        [(Assignment
                                            (ArrayItem
                                                (Var 223 b)
                                                [(()
                                                (Var 223 i)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (RealConstant
                                                5.000000
                                                (Real 8)
                                            )
                                            ()
                                            .false.
                                        )]
                                        []
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 223 i)
                                        (IntegerConstant 0 (Integer 4) Decimal)
                                        (IntegerConstant 1151 (Integer 4) Decimal)
                                        (IntegerConstant 1 (Integer 4) Decimal))
                                        [(SubroutineCall
                                            223 vector_copy_f64[9216]f64[9216]i32@IntrinsicOptimization
                                            ()
                                            [((Var 223 a))
                                            ((Var 223 b))
                                            ((IntegerBinOp
                                                (Var 223 i)
                                                Mul
                                                (IntegerConstant 8 (Integer 4) Decimal)
                                                (Integer 4)
                                                ()
                                            ))
                                            ((IntegerBinOp
                                                (IntegerBinOp
                                                    (Var 223 i)
                                                    Add
                                                    (IntegerConstant 1 (Integer 4) Decimal)
                                                    (Integer 4)
                                                    ()
                                                )
                                                Mul
                                                (IntegerConstant 8 (Integer 4) Decimal)
                                                (Integer 4)
                                                ()
                                            ))
                                            ((IntegerConstant 1 (Integer 4) Decimal))
                                            ((IntegerConstant 8 (Integer 4) Decimal))]
                                            ()
                                        )]
                                        []
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 223 i)
                                        (IntegerConstant 0 (Integer 4) Decimal)
                                        (IntegerBinOp
                                            (IntegerConstant 9216 (Integer 4) Decimal)
                                            Sub
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            (Integer 4)
                                            (IntegerConstant 9215 (Integer 4) Decimal)
                                        )
                                        (IntegerConstant 1 (Integer 4) Decimal))
                                        [(Assert
                                            (RealCompare
                                                (ArrayItem
                                                    (Var 223 a)
                                                    [(()
                                                    (Var 223 i)
                                                    ())]
                                                    (Real 8)
                                                    RowMajor
                                                    ()
                                                )
                                                Eq
                                                (RealConstant
                                                    5.000000
                                                    (Real 8)
                                                )
                                                (Logical 4)
                                                ()
                                            )
                                            ()
                                        )]
                                        []
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    ()
                    [numpy]
                    .false.
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        228
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    228
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        228 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
